src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -461,8 +461,19 @@ namespace OpenMS
         {
           QMessageBox::critical(this, "Error", QString("Error determining output type from tool. Tool is not compatible with TOPPView"));
         }
-          // If we have multiple valid output types, we don't know what the file actually contains, so use UNKNOWN
-
+      else
+      // If we have multiple valid output types, check if out_type exists and is specified.
+      {
+        if (tool_params_.exists("out_type") && tool_params_.getValue("out_type") != "")
```

Comment:
this is probably too specific.
Something like `output_combo_->currentText().toStdString() + "_type"` will probably work in general.